--- origsrc/audacious-plugins-2.4.3/src/ffaudio/Makefile	2011-01-12 05:13:26.000000000 -0600
+++ src/audacious-plugins-2.4.3/src/ffaudio/Makefile	2011-02-03 03:33:02.976887300 -0600
@@ -10,4 +10,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS} ${SIMD_CFLAGS} ${FFMPEG_CFLAGS} -I../.. -std=c99 ${GCC42_CFLAGS} -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} ${FFMPEG_LIBS} -laudtag
+LIBS := -laudtag ${LIBS} ${GTK_LIBS} ${GLIB_LIBS} ${FFMPEG_LIBS}
--- origsrc/audacious-plugins-2.4.3/src/mpg123/Makefile	2011-01-12 05:13:26.000000000 -0600
+++ src/audacious-plugins-2.4.3/src/mpg123/Makefile	2011-02-03 03:32:19.044273800 -0600
@@ -31,4 +31,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS} ${MOWGLI_CFLAGS} ${ARCH_DEFINES} ${SIMD_CFLAGS} -I../..
 LDFLAGS += ${AUDLDFLAGS}
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} ${MOWGLI_LIBS} -laudtag -lm
+LIBS := -laudtag ${LIBS} ${GTK_LIBS} ${GLIB_LIBS} ${MOWGLI_LIBS} -lm
--- origsrc/audacious-plugins-2.4.3/src/wavpack/Makefile	2011-01-12 05:13:26.000000000 -0600
+++ src/audacious-plugins-2.4.3/src/wavpack/Makefile	2011-02-03 03:33:12.861567100 -0600
@@ -9,4 +9,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS} ${WAVPACK_CFLAGS} -I../..
-LIBS += ${WAVPACK_LIBS} ${GTK_LIBS} ${GLIB_LIBS} -laudtag
+LIBS := -laudtag ${LIBS} ${WAVPACK_LIBS} ${GTK_LIBS} ${GLIB_LIBS}
