--- origsrc/audacious-plugins-3.0.4/src/amidi-plug/i_backend.c	2011-10-20 17:21:19.000000000 -0500
+++ src/audacious-plugins-3.0.4/src/amidi-plug/i_backend.c	2011-10-23 21:56:46.565747200 -0500
@@ -49,7 +49,7 @@ GSList * i_backend_list_lookup( void )
     while ( backend_directory_entry != NULL )
     {
       /* simple filename checking */
-      if ( i_str_has_pref_and_suff( backend_directory_entry , "ap-" , ".so" ) == TRUE )
+      if ( i_str_has_pref_and_suff( backend_directory_entry , "ap-" , "."G_MODULE_SUFFIX ) == TRUE )
       {
         GModule * module;
         gchar * (*getapmoduleinfo)( gchar ** , gchar ** , gchar ** , gint * );
@@ -110,7 +110,7 @@ void i_backend_list_free( GSList * backe
 
 gint i_backend_load( gchar * module_name )
 {
-  gchar * module_pathfilename = g_strjoin( "" , AMIDIPLUGBACKENDDIR , "/ap-" , module_name , ".so" , NULL );
+  gchar * module_pathfilename = g_strjoin( "" , AMIDIPLUGBACKENDDIR , "/ap-" , module_name , "."G_MODULE_SUFFIX , NULL );
   DEBUGMSG( "loading backend '%s'\n" , module_pathfilename );
   backend.gmodule = g_module_open (module_pathfilename, G_MODULE_BIND_LAZY |
    G_MODULE_BIND_LOCAL);
