--- origsrc/audacious-plugins-3.0.4/src/amidi-plug/i_backend.c	2011-10-20 17:21:19.000000000 -0500
+++ src/audacious-plugins-3.0.4/src/amidi-plug/i_backend.c	2011-10-23 21:56:46.565747200 -0500
@@ -49,7 +49,7 @@ GSList * i_backend_list_lookup( void )
     while ( backend_directory_entry != NULL )
     {
       /* simple filename checking */
-      if ( i_str_has_pref_and_suff( backend_directory_entry , "ap-" , ".so" ) == TRUE )
+      if ( i_str_has_pref_and_suff( backend_directory_entry , "ap-" , "."G_MODULE_SUFFIX ) == TRUE )
       {
         GModule * module;
         gchar * (*getapmoduleinfo)( gchar ** , gchar ** , gchar ** , gint * );
@@ -113,1 +113,1 @@ void i_backend_list_free( GSList * backe
-    module_pathfilename = g_strjoin ("", AMIDIPLUGBACKENDDIR, "/ap-", module_name, ".so", NULL);
+    module_pathfilename = g_strjoin ("", AMIDIPLUGBACKENDDIR, "/ap-", module_name, "." G_MODULE_SUFFIX, NULL);
